File: skellig-test-step-processing/src/main/java/org/skellig/teststep/processing/exception/TestStepProcessingException.java
package org.skellig.teststep.processing.exception;

public class TestStepProcessingException extends RuntimeException {

    public TestStepProcessingException(String message) {
        super(message);
    }

    public TestStepProcessingException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: skellig-test-step-processing/src/main/kotlin/org/skellig/teststep/processing/exception/TestStepProcessingException.kt
package org.skellig.teststep.processing.exception

/**
 * Represents an exception that occurs during the processing of a test step by
 * [TestStepProcessor][org.skellig.teststep.processing.processor.TestStepProcessor].
 *
 * @param message The detail message of the exception.
 * @param cause The cause of the exception.
 */
class TestStepProcessingException(message: String?, cause: Throwable?) : RuntimeException(message, cause) {

    constructor(message: String?) : this(message, null)
}